Edits to five short indexing pages as part of site review (#201)
pinot-contrib/pinot-docs: basics/indexing/bloom-filter.md

---
description: This page describes configuring the bloom filter for Apache Pinot
---

> # Bloom Filter

Reviewer: ```suggestion
# Bloom filter
```

The bloom filter prunes segments that do not contain any record matching an **EQUALITY** predicate.

This is useful for a query like the following:

```sql
SELECT COUNT(*) 
FROM baseballStats 
WHERE playerID = 12345
```

> There are 3 parameters to configure the bloom filter:

Reviewer: Good example of where we can start making the structure more clear and modular.
For each index doc, we can add a procedural header for "Configure the xyz filter" and rewrite explicit steps to complete.
In future, we could regroup similar content into another chunk, perhaps "Configure the table index" where it makes sense to do so

Reviewer: 1. Set the false positive probability `fpp` for the bloom filter between `0` to `1` (`0.05` by default). As the `fpp` decreases, the bloom filter size and accuracy increases.
...

> * **`fpp`**: False positive probability of the bloom filter (from `0` to `1`, `0.05` by default). The lower the `fpp` , the higher accuracy the bloom filter has, but it will also increase the size of the bloom filter.

Reviewer: ```suggestion
* **`fpp`**: False positive probability of the bloom filter (from `0` to `1`, `0.05` by default). A lower `fpp` increase both accuracy and size of the bloom filter.
```

> * **`maxSizeInBytes`**: Maximum size of the bloom filter (unlimited by default). If a `fpp` setting generates a bloom filter larger than this size, using this setting will increase the `fpp` to keep the bloom filter size within this limit.

Reviewer: 2. Set the maximum size of the bloom filter `maxSizeInBytes` (unlimited by default)....

> * **`loadOnHeap`**: Whether to load the bloom filter using heap memory or off-heap memory (`false` by default).

Reviewer: 3. Specify whether to load the bloom filter....
